llvm/include/llvm/Support/FormatProviders.h

```diff
@@ -132,11 +132,10 @@ struct format_provider<
 private:
 public:
   static void format(const T &V, llvm::raw_ostream &Stream, StringRef Style) {
-    HexPrintStyle HS;
     size_t Digits = 0;
-    if (consumeHexStyle(Style, HS)) {
-      Digits = consumeNumHexDigits(Style, HS, 0);
-      write_hex(Stream, V, HS, Digits);
+    if (const auto &HS = consumeHexStyle(Style)) {
```

Comment:
use non-ref (best to avoid reference lifetime extension) (and probably non-const, once it's local - there aren't many uses of local const in LLVM) here and below. Maybe name the type rather than using `auto`